core/textile/client.go

```diff
@@ -313,3 +334,30 @@ func (tc *textileClient) getThreadContext(parentCtx context.Context, threadName
 
 	return ctx, nil
 }
+
+// Checks for connection and initialization needs.
+func (tc *textileClient) healthcheck(ctx context.Context) {
+	log.Debug("Textile Client healthcheck... Start.")
+
+	if tc.isInitialized == false {
+		tc.initialize(ctx)
+	}
+
+	tc.ConnectToHub(ctx)
```

Comment:
Should we wrap this in a `tc.isConnectedToHub == false` clause?

Reply:
Updated `ConnectToHub` to better reflect what it's doing right now. So now it's called `checkHubConnection` and it will get the token from the store, refresh it if necessary, and connect to mailbox if necessary. If any of these steps fail, `tc.isConnectedToHub` will be set to `false`.